src/gluonnlp/data/transforms.py

```diff
@@ -1280,3 +1286,171 @@ def _truncate_seq_pair(self, tokens_a, tokens_b, max_length):
+            _url_format = '{repo_url}gluon/dataset/vocab/{file_name}'
+            download(_url_format.format(repo_url=repo_url, file_name=archive_name),
+                     path=zip_file_path,
+                     sha1_hash=archive_hash,
+                     overwrite=True)
+            with zipfile.ZipFile(zip_file_path) as zf:
+                if not os.path.exists(file_path):
+                    zf.extractall(root)
+            try:
+                os.remove(zip_file_path)
+            except OSError as e:
+                # file has already been removed.
+                if e.errno == 2:
+                    pass
+                else:
+                    raise e
+
+            if check_sha1(file_path, sha1_hash):
+                self._read_bpe_ranks(file_path)
+            else:
+                raise ValueError('Downloaded file has different hash. Please try again.')
+        self._cache = {}
+        self._token_pattern = re.compile(
+            r'\'s|\'t|\'re|\'ve|\'m|\'ll|\'d| ?\p{L}+'
+            r'| ?\p{N}+| ?[^\s\p{L}\p{N}]+|\s+(?!\S)|\s+')
+
+    def _read_bpe_ranks(self, file_path):
+        with io.open(file_path, 'r', encoding='utf-8') as f:
+            bpe_data = f.read()
+            self._bpe_ranks = {
+                tuple(merge_str.split()): i for i, merge_str
+                in enumerate(bpe_data.split('\n')[1:-1])}
+
+    def get_bpe_subword(self, token):
+        """ Encode the word token into BPE subwords
+
+        Parameters
+        ----------
+        token : str
+
+        Returns
+        -------
+        chars : list(str)
+        """
+        if token in self._cache:
+            return self._cache[token]
+        chars = list(token)
+        while len(chars) > 0:
+            min_pair, min_rank = None, float('inf')
+            # Find the pair with the minimum rank
+            for i in range(1, len(chars)):
+                pair = (chars[i - 1], chars[i])
+                rank = self._bpe_ranks.get(pair, float('inf'))
+                if rank < min_rank:
+                    min_rank = rank
+                    min_pair = pair
+            if min_pair is None or min_pair not in self._bpe_ranks:
+                break
+            # Merge the pair
+            last, tail = chars[0], 1
+            for index in range(1, len(chars)):
+                if (last, chars[index]) == min_pair:
+                    chars[tail - 1] = last + chars[index]
+                    last = last + chars[index]
+                else:
+                    chars[tail - 1] = last
+                    tail += 1
+                    last = chars[index]
+            chars[tail - 1] = last
+            chars = chars[:tail]
+        self._cache[token] = chars
+        return chars
+
+    def __call__(self, sample):
+        """
+
+        Parameters
+        ----------
+        sample : str
+
+        Returns
+        -------
+        ret : list(str)
+        """
+        ret = []
```

Comment:
Is this py3 specific? Should we add some python version check in the constructor?